### ISSUE TITLE
Error running Multik 0.2.0-dev-5 on Github Actions with ubuntu-latest

### ISSUE THREAD

**Issue by breandan** [Kotlin/multik#121]
It looks like you are preparing for a new release of multik. Tests pass locally on MacOS/ARM64, but I encountered the following error when trying `0.2.0-dev-5` on Github's CI server:

```
java.lang.UnsatisfiedLinkError: /tmp/jni_multik7503611758506959258/libmultik_jni-linuxX64.so: /tmp/jni_multik7503611758506959258/libmultik_jni-linuxX64.so: ELF load command address/offset not properly aligned
            at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
            at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
            at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
            at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
            at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2389)
            at java.base/java.lang.Runtime.load0(Runtime.java:755)
            at java.base/java.lang.System.load(System.java:1953)
            at org.jetbrains.kotlinx.multik.openblas.JvmLoader.load(Loader.kt:72)
            at org.jetbrains.kotlinx.multik.openblas.JvmNativeEngine.<init>(JvmNativeEngine.kt:7)
            at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
            at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
            at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
            at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
            at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
            at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:789)
```

These are the versions I am currently using:

```
        val multikVersion = "0.2.0-dev-5"
        implementation("org.jetbrains.kotlinx:multik-core:$multikVersion")
//  implementation("org.jetbrains.kotlinx:multik-core-linuxx64:$multikVersion")
//  implementation("org.jetbrains.kotlinx:multik-kotlin-linuxx64:$multikVersion")
        implementation("org.jetbrains.kotlinx:multik-core-jvm:$multikVersion")
        implementation("org.jetbrains.kotlinx:multik-default:$multikVersion")
        implementation("org.jetbrains.kotlinx:multik-openblas-jvm:$multikVersion")
        implementation("org.jetbrains.kotlinx:multik-openblas:$multikVersion")
//  implementation("org.jetbrains.kotlinx:multik-native:$multik_version")
```

When I added `multik-core-linuxx64`, I received the following error:

```
Execution failed for task ':kspKotlinJvm'.
> Error while evaluating property 'filteredArgumentsMap' of task ':kspKotlinJvm'
   > Could not resolve all files for configuration ':jvmCompileClasspath'.
      > Could not resolve org.jetbrains.kotlinx:multik-kotlin-linuxx64:0.2.0-dev-5.
        Required by:
            project :
         > No matching variant of org.jetbrains.kotlinx:multik-kotlin-linuxx64:0.2.0-dev-5 was found. The consumer was configured to find an API of a library, preferably optimized for standard JVMs, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
             - Variant 'linuxX64ApiElements-published' capability org.jetbrains.kotlinx:multik-kotlin-linuxx64:0.2.0-dev-5 declares a library:
                 - Incompatible because this component declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native' and the consumer needed an API of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm'
                 - Other compatible attribute:
                     - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
```

I am currently unsure how to proceed, any feedback you might have would be appreciated. Thanks!

**Comment by devcrocod** [comment]
> Tests pass locally on MacOS/ARM64, but I encountered the following error when trying 0.2.0-dev-5 on Github's CI server:

What is the actual version of Ubuntu?
There is a known bug with Ubuntu 18.04 and older ([link1](https://github.com/Kotlin/multik/tree/v0.2.0#support-platforms), #107, #98). I tested on 20.04, 22.04, and multik ran without errors.

> When I added multik-core-linuxx64

You have added a native artifact to jvm dependency. These artifacts are needed for [multiplatform projects](https://kotlinlang.org/docs/multiplatform-add-dependencies.html). 

For jvm the following dependency:

```
implementation("org.jetbrains.kotlinx:multik-core:$multikVersion")
```

or

```
implementation("org.jetbrains.kotlinx:multik-core-jvm:$multikVersion")
```

**Comment by breandan** [comment]
Hi @devcrocod, thank you for your reply! I tried both of those options, however am still receiving the aforementioned errors. In case it helps, here are the full stack traces for the [JVM](https://gist.github.com/breandan/26281bce64085bb4d3e406a951c46b61) and the [multiplatform](https://gist.github.com/breandan/191b3b98d68fc91dbebfb8f9a8944d71) dependencies. It looks like [`ubuntu-latest`](https://github.com/actions/virtual-environments#available-environments) is currently on `20.04`.

**Comment by breandan** [comment]
I am using [Kotlin/JS with the IR compiler](https://github.com/breandan/galoisenne/blob/0c925319242f3c000cccaa8ac468cb4aca514e20/build.gradle.kts#L88), however this error occurs in a [JVM test](https://github.com/breandan/galoisenne/blob/master/src/jvmTest/kotlin/ai/hypergraph/markovian/mcmc/MarkovChainTest.kt). Strangely, it does not occur locally on `macosArm64`, only `linuxX64` on the build server.

**Comment by devcrocod** [comment]
I took your repository `galoisenne` and tested. Tested on macosX64, Ubuntu 20.04, Ubuntu 22.04. On macos and ubuntu 22 all tests passed. 

Native library bug for dependency jvm was reproduced on ubuntu 20.04. I have already made changes to fix this error. The fix is already expected in the 0.2.0 release early next week.

There was an error for JS, as I said about the IR compiler. Fixed when I added `JS (IR)`. Fixed when I added JS. After that, all the tests in my docker passed, except for those that require a browser.
